### ISSUE TITLE
Unexpected bicep type: 'secureString'

### ISSUE THREAD

**Issue by zedy-wj** [Azure/azure-dev#1289]
**Describe the issue:**
We are applying the creation of `APIM` to all templates and testing them. But when we tested `todo-csharp-sql` and `todo-csharp-sql-swa-func` templates, we found this problem when executing `azd up`:  Unexpected bicep type: 'secureString'. According to the investigation, this is a problem in `staging` branch, and has nothing to do with `APIM`.

**Repro steps:**
1. Login with `azd login`
2. Run command: `azd init -t Azure-Samples/todo-csharp-sql -b pr/1279`
3. Run command: `azd up`
![image](https://user-images.githubusercontent.com/81678720/208613180-353506ab-e7f3-4ccb-90fa-7ae64c1afa49.png)

**Environments:**
OS: Codespaces, Windows desktop, MacOS desktop, Linux desktop, Devcontainer in VS Code, WSL
Templates: todo-csharp-sql and todo-csharp-sql-swa-func
Branch: [pr/1279](https://github.com/Azure/azure-dev/pull/1279)
Azd version: 0.4.0-beta.2-daily.2062439 (commit e9baa6ae0f948aaa4fa58bfc9cf0160eff2de275)

**Expected behavior:**
`azd up` can create and deploy resources normally.

@rajeshkamal5050 for notification.

**Comment by ellismg** [comment]
I opened #1294 to address this issue.  CI should have privates soon.  I was able verify locally that I was able to deploy the given infrastructure.

**Comment by v-jiaodi** [comment]
@rajeshkamal5050 This issue also occurs in the following cases:


- OS: WSL, MacOS desktop, Linux desktop, Devcontainer in VS Code, Codespaces, Windows desktop.
- Template: todo-csharp-sql and todo-csharp-sql-swa-func
- Branch: Staging
- Azd version: 0.4.0-beta.2-daily.2081395  (commit https://github.com/Azure/azure-dev/commit/852cee7207fd484858205c0bab7de22a13740e83)

**Comment by v-jiaodi** [comment]
@ellismg  Similar issue occurs in the following cases, please help to review and resolve it.
- **OS:** Windows desktop.
- **Template:** todo-csharp-sql, todo-csharp-sql-swa-func, https://github.com/pamelafox/flask-gallery-container-app
- **Azd version:** 0.6.0-beta.2  (commit https://github.com/Azure/azure-dev/commit/c4a201e55a13eba77f9037b296a90e0f81b00dbe)

**Error message**:
![image](https://user-images.githubusercontent.com/80496810/219609319-70889e57-d44c-43f2-83d7-a8c117a34750.png)

**Repro steps**:

1. Login with `azd login`
2. Run command: `azd init -t Azure-Samples/todo-csharp-sql` 
3. Run command: `azd up`

**Comment by v-jiaodi** [comment]
@ellismg  Same issue occurs in the following cases, please help to review and resolve it.
- **OS:** Windows desktop.
- **Template:** https://github.com/Azure-Samples/apptemplate-wordpress-on-ACA
- **Azd version:** 0.6.0-beta.2  (commit https://github.com/Azure/azure-dev/commit/c4a201e55a13eba77f9037b296a90e0f81b00dbe)

**Error message**:
![image](https://user-images.githubusercontent.com/80496810/221823124-a247ed47-5adc-48cf-9f49-bcf8b959e818.png)

**Repro steps**:

1.Clone this repository localy.
2. Login with `azd login`
3. Run command: `azd init` 
4. Add the following environment variables to `.env`.
```
AZD_PIPELINE_PROVIDER="github"
AZURE_ADMIN_PASSWORD="<THE JUMPHOST PASSWORD>"
AZURE_ADMIN_USERNAME="<THE JUMPHOST USERNAME>"
AZURE_APPLICATION_NAME="<AN APPLICATION NAME>"
AZURE_ENV_NAME="prod"
AZURE_FQDN="<THE FQDN OF THE SITE>"
AZURE_LOCATION="<AN AZURE DC REGION>"
AZURE_MARIADB_PASSWORD="<THE MARIADB PASSWORD>"
AZURE_PRINCIPAL_ID=""
AZURE_SUBSCRIPTION_ID="<YOUR AZURE SUBSCRIPTION ID>"
resourceGroupName="<THE RESOURCE GROUP NAME>"
```
5. Run `azd provision`.

**Comment by weikanglim** [comment]
@v-jiaodi There's a slight difference between this new issue and the old issue.

In the new error, the message reads: unexpected bicep type: '`securestring`' instead of `secureString`

This has been fixed in the latest `daily` build, after #1650. Can you test it on `daily` and verify that it is fixed?

**Comment by weikanglim** [comment]
Tested with the `daily`, and everything works fine.

**Comment by v-jiaodi** [comment]
Tested with the `daily`, this issue no longer exists.
